core-httpclient-impl/src/main/java/com/optimizely/ab/odp/DefaultODPApiManager.java

```diff
@@ -135,7 +135,11 @@ String getSegmentsStringForRequest(Set<String> segmentsList) {
     public String fetchQualifiedSegments(String apiKey, String apiEndpoint, String userKey, String userValue, Set<String> segmentsToCheck) {
         HttpPost request = new HttpPost(apiEndpoint);
         String segmentsString = getSegmentsStringForRequest(segmentsToCheck);
-        String requestPayload = String.format("{\"query\": \"query {customer(%s: \\\"%s\\\") {audiences(subset: [%s]) {edges {node {name state}}}}}\"}", userKey, userValue, segmentsString);
+
+        String query = String.format("query($userId: String, $audiences: [String]) {customer(%s: $userId) {audiences(subset: $audiences) {edges {node {name state}}}}}", userKey);
+        String variables = String.format("{\"userId\": \"%s\", \"audiences\": [%s]}", userValue, segmentsString);
+        String requestPayload = String.format("{\"query\": \"%s\", \"variables\": %s}", query, variables);
```

Comment:
Looks like no gain at all in Java with this new format. I guess you're trying to avoid use JSON parser here :)

Reply:
There is still a gain. We are now using graphQL json format which takes variables separately which means if someone wants to inject anything in the query, it wont work because we are not concatenating variables directly in the query, we are passing them as variables.